resource/k8s/configmapresource/delete.go

```diff
@@ -27,13 +27,21 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 		} else {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted ConfigMap %#q in namespace %#q", configMap.Name, configMap.Namespace))
 		}
-
 	}
 
 	return nil
 }
 
-func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desiredState interface{}) ([]*corev1.ConfigMap, error) {
+func (r *Resource) newDeleteChangeForDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) ([]*corev1.ConfigMap, error) {
```

Comment:
I think we can leave `desiredState` from parameters, right?

Reply:
In theory yes. I would like to keep the interface as it is though since this is what we do everywhere. I do not want to promote snowflakes. When this is something we want to do in general we should get to an agreement. Feel free to push this.